pkg/runner/cluster_k8s.go

```diff
@@ -293,138 +293,60 @@ func (*ClusterK8sRunner) ID() string {
-		return
-	}
-	return
-}
-
-func (c *ClusterK8sRunner) healthcheckEFS() (efsCheck api.HealthcheckItem) {
-	efsCheck = api.HealthcheckItem{Name: "efs", Status: api.HealthcheckStatusFailed, Message: "efs provisioner is not running"}
-
-	client := c.pool.Acquire()
-	defer c.pool.Release(client)
-
-	pods, err := client.CoreV1().Pods(c.config.Namespace).List(metav1.ListOptions{
-		LabelSelector: "app=efs-provisioner",
-	})
-	if err != nil {
-		efsCheck.Message = err.Error()
-		return
-	}
-	if len(pods.Items) != 1 {
-		efsCheck.Message = fmt.Sprintf("expected 1 EFS provisioner pod. found %d.", len(pods.Items))
-		return
-	}
-
-	pod := pods.Items[0]
-	if pod.Status.Phase != "Running" {
-		return
-	}
-
-	efsCheck = api.HealthcheckItem{Name: "efs", Status: api.HealthcheckStatusOK, Message: "efs provisioner is running"}
-	return
-}
-
-func (c *ClusterK8sRunner) healthcheckRedis() (redisCheck api.HealthcheckItem) {
-	redisCheck = api.HealthcheckItem{Name: "redis", Status: api.HealthcheckStatusFailed, Message: "redis service is not running"}
-
-	client := c.pool.Acquire()
-	defer c.pool.Release(client)
-
-	pods, err := client.CoreV1().Pods(c.config.Namespace).List(metav1.ListOptions{
-		LabelSelector: "app=redis",
-	})
-	if err != nil {
-		redisCheck.Message = err.Error()
-		return
-	}
-	if len(pods.Items) != 1 {
-		redisCheck.Message = fmt.Sprintf("expected 1 redis pod. found %d.", len(pods.Items))
-		return
-	}
-
-	for _, pod := range pods.Items {
-		if pod.Status.Phase != "Running" {
-			return
-		}
-	}
-
-	redisCheck = api.HealthcheckItem{Name: "redis", Status: api.HealthcheckStatusOK, Message: "redis service is running"}
-	return
-}
-
-func (c *ClusterK8sRunner) healthcheckSidecar() (sidecarCheck api.HealthcheckItem) {
-	sidecarCheck = api.HealthcheckItem{Name: "sidecar", Status: api.HealthcheckStatusFailed, Message: "sidecar service is not reachable"}
+func (c *ClusterK8sRunner) Healthcheck(ctx context.Context, engine api.Engine, ow *rpc.OutputWriter, fix bool) (*api.HealthcheckReport, error) {
+	// TODO how does one pass the context to k8s API calls?
 
-	// get number of worker nodes
+	c.initPool()
 	client := c.pool.Acquire()
-	defer c.pool.Release(client)
 
+	// How many plan worker nodes are there?
 	res, err := client.CoreV1().Nodes().List(metav1.ListOptions{
 		LabelSelector: "testground.nodetype=plan",
 	})
 	if err != nil {
-		sidecarCheck.Message = err.Error()
-		return
+		return nil, err
 	}
+	planNodes := res.Items
 
-	nodes := len(res.Items)
+	hh := &hc.Helper{}
 
-	pods, err := client.CoreV1().Pods(c.config.Namespace).List(metav1.ListOptions{
-		LabelSelector: "name=testground-sidecar",
-	})
-	if err != nil {
-		sidecarCheck.Message = err.Error()
-		return
-	}
-	if len(pods.Items) != nodes {
-		sidecarCheck.Message = fmt.Sprintf("expected %d sidecar pods. found %d.", nodes, len(pods.Items))
-		return
-	}
+	hh.Enlist("kops validate",
```

Comment:
We should do this healthcheck in a different way - just check that we have a number of healthy nodes maybe?

It'd be nice if we don't make assumptions on the Kubernetes cluster in Testground - users should be able to spin it up as they see fit - for example if there is a way to run the networking part in EKS or GKE, it should be possible to run Testground there as well.

Reply:
That could very well be off scope for this PR, so just ignore if you don't want to fix now!